lib/Browser.js

```diff
@@ -30,6 +30,55 @@ class Browser extends EventEmitter {
     this._screenshotTaskQueue = new TaskQueue();
     this._connection = connection;
     this._closeCallback = closeCallback || new Function();
+    /** @type {Map<string, Promise<Page>>} */
+    this._pages = new Map();
+    this._connection.on('Target.targetCreated', this._targetCreated.bind(this));
+    this._connection.on('Target.targetDestroyed', this._targetDestroyed.bind(this));
+  }
+
+  /**
+   * @param {!Connection} connection
+   * @param {boolean} ignoreHTTPSErrors
+   * @param {function()=} closeCallback
+   */
+  static async create(connection, ignoreHTTPSErrors, closeCallback) {
+    const browser =  new Browser(connection, ignoreHTTPSErrors, closeCallback);
+    await connection.send('Target.setDiscoverTargets', {discover: true});
```

Comment:
You should call `Target.getTargets` to read present state and also you should modify it do the same thing `Target.setDiscoverTargets` does. That way you don't have a race between these calls and target creation.

Reply:
Please don't :) Target.setDiscoverTargets should notify about all existing ones through Target.targetCreated. There is no need to use Target.getTargets at all.

Reply:
Well, if you look at [TargetHandler::SetDiscoverTargets](https://cs.chromium.org/chromium/src/content/browser/devtools/protocol/target_handler.cc?q=TargetHandler::SetD&sq=package:chromium&l=187), you would be certain it does not. Except for I now see that your observer has side effects. But that is just mean, you know, calling it observer and making it change the model!